### ISSUE TITLE
Feature request, using Paddle OCR offline engine

### ISSUE THREAD

**Issue by cary-rowen** [sharkboyto/nao#28]
There is a open source project called Paddle OCR. This project supports multiple languages. In the Chinese and English scenarios I tested, its recognition rate exceeded Windows OCR.
The only thing you might have to think about is how to compile an offline version for x86 to use it in NVDA addon.
Screen readers in China have distributed it as part of their screen readers, and so far everything seems to be working fine.

Paddle OCR: https://github.com/PaddlePaddle/PaddleOCR/

**Comment by davidedec** [comment]
Hi @cary-rowen 
Thank you for your suggestion, we took some time to analyze the PaddleOCR project and looks interesting.
However, like others offline ocr engines we tested, we first needs to implement some sort of "dependency download" system in NAO. It is not reasonable to grow hundreds of megabytes the plugin package for each engine or tool we will need.
This kind of system is in our TODO list and as soon as we will have, we will probably add also Paddle OCR to the "optional downloadable" engines.
Thank you again for your support!

**Comment by cary-rowen** [comment]
Hi @davidedec 

Thanks for the investigation, it's interesting, as you said it would be nice to implement a mechanism for users to download only the languages ​​they have available.

Grateful

**Comment by cary-rowen** [comment]
HI, @davidedec 

Any updates on this feature request?

Best

**Comment by cary-rowen** [comment]
Hi,
I think this project can make it easy for NAO to integrate PaddleOCR:
https://github.com/hiroi-sora/PaddleOCR-json

**Comment by dreinn** [comment]
Hello cary-rowen, thank you very  much for that resource!
do you know if this OCR is able to recognize also various languages present in India? We received a request also for that, but unfortunately Windows OCR is not able to do it. Have you some information about this? 
I see that a model called devanagari is available, it should be a particular manner to write charachters, so I think it could be possible.

**Comment by cary-rowen** [comment]
I think it's ok.
PaddleOCR aims to create a rich, leading, and practical OCR toolkit, which not only provides Chinese and English models for general scenarios but also provides models trained specifically for English scenarios, as well as models for 80 less commonly used languages.

**Comment by dreinn** [comment]
Hello, we took some time to study the situation and unfortunately, based on the tests we have done, we are unable to integrate PaddleOcr as it takes far too long to perform a scan. Especially on systems that do not have a GPU, it is not acceptable to have to wait for minutes and minutes. I don't know if a lighter version has been installed in Chinese screen readers, but PaddleOCR takes a long time to initialize, and there are many people who do not have a dedicated video card. Anyway, we have decided to follow your idea. We will not introduce PaddleOCR, but we will definitely include another OCR that supports Chinese and other Eastern languages well, which Windows OCR does not handle. Thank you for providing us with an interesting idea. We will be able to improve Nao, even if we do not use PaddleOCR.
